cloud/test/meta_service_operation_log_test.cpp

```diff
@@ -355,4 +355,309 @@ TEST(MetaServiceOperationLogTest, DropPartitionLog) {
+        ret->first = instance_id;
+        ret->second = true;
+    });
+    sp->enable_processing();
+
+    constexpr int64_t db_id = 123;
+    constexpr int64_t table_id = 10001;
+    constexpr int64_t index_id = 10002;
+
+    {
+        // write instance
+        InstanceInfoPB instance_info;
+        instance_info.set_instance_id(instance_id);
+        instance_info.set_multi_version_status(MULTI_VERSION_WRITE_ONLY);
+        std::unique_ptr<Transaction> txn;
+        ASSERT_EQ(meta_service->txn_kv()->create_txn(&txn), TxnErrorCode::TXN_OK);
+        txn->put(instance_key(instance_id), instance_info.SerializeAsString());
+        ASSERT_EQ(txn->commit(), TxnErrorCode::TXN_OK);
+
+        meta_service->resource_mgr()->refresh_instance(instance_id);
+        ASSERT_TRUE(meta_service->resource_mgr()->is_version_write_enabled(instance_id));
+    }
+
+    {
+        // Prepare index 0,1,2,3
+        brpc::Controller ctrl;
+        IndexRequest req;
+        IndexResponse res;
+        req.set_db_id(db_id);
+        req.set_table_id(table_id);
+        req.add_index_ids(index_id);
+        req.add_index_ids(index_id + 1);
+        req.add_index_ids(index_id + 2);
+        req.add_index_ids(index_id + 3);
+        meta_service->prepare_index(&ctrl, &req, &res, nullptr);
+        ASSERT_EQ(res.status().code(), MetaServiceCode::OK) << res.status().DebugString();
+    }
+
+    {
+        // Commit index 2,3
+        brpc::Controller ctrl;
+        IndexRequest req;
+        IndexResponse res;
+        req.set_db_id(db_id);
+        req.set_table_id(table_id);
+        req.add_index_ids(index_id + 2);
+        req.add_index_ids(index_id + 3);
+        meta_service->commit_index(&ctrl, &req, &res, nullptr);
+        ASSERT_EQ(res.status().code(), MetaServiceCode::OK) << res.status().DebugString();
+    }
+
+    auto txn_kv = meta_service->txn_kv();
+    size_t num_logs = count_range(txn_kv.get(), versioned::log_key(instance_id),
+                                  versioned::log_key(instance_id) + "\xFF");
+
+    {
+        // Drop index 0
+        brpc::Controller ctrl;
+        IndexRequest req;
+        IndexResponse res;
+        req.set_db_id(db_id);
+        req.set_table_id(table_id);
+        req.add_index_ids(index_id);
+        meta_service->drop_index(&ctrl, &req, &res, nullptr);
+        ASSERT_EQ(res.status().code(), MetaServiceCode::OK) << res.status().DebugString();
+
+        // No operation log are generated for drop index 0 (it is not committed).
+        size_t new_num_logs = count_range(txn_kv.get(), versioned::log_key(instance_id),
+                                          versioned::log_key(instance_id) + "\xFF");
+        ASSERT_EQ(new_num_logs, num_logs)
+                << "Expected no new operation logs for drop index 0, but found "
+                << new_num_logs - num_logs << dump_range(txn_kv.get());
+    }
+
+    {
+        // Drop index 1,2, it should generate operation logs.
```

Comment:
Please verify that the recycle index key does not exist if the operation log is generated.